### ISSUE TITLE
python310 on darwin does not support macosx_11_0 tags

### ISSUE THREAD

**Issue by maksbotan** [NixOS/nixpkgs#229864]
### Describe the bug

`nixpkgs.python310` with `nixpkgs.python310Packages.pip` refuse to install to install wheels that have `cp310-cp310-macosx_11_0_x86_64` tag.

### Steps To Reproduce
1. `niv-env -i nixpkgs.python310 nixpkgs.python310Packages.pip`
2. `python -m pip debug --verbose | grep cp310-cp310-macosx_11_0_x86_64`

With this command I get an empty output. If I grep for `cp310-cp310-macosx`, I see that only `10_x` tags are supported.

Therefore, when I try to install a wheel built with this tag, e.g. https://files.pythonhosted.org/packages/6a/4c/34ee83616714e6194980e0429a8d58746f011848ea766cd61083a6ae5535/SQLAlchemy-1.4.46-cp310-cp310-macosx_11_0_x86_64.whl, pip throws an error:

```
ERROR: SQLAlchemy-1.4.46-cp310-cp310-macosx_11_0_x86_64.whl is not a supported wheel on this platform.
```

Can nixpkgs python310 be updated to support macosx >10?

### Expected behavior

Tag is present in the output.

### Additional context

I'm on macOS Monterey 12.6 on MacBook with M1 cpu. I run nix through Rosetta, so the architecture is `x86_64`.

```
$ sw_vers 
ProductName:	macOS
ProductVersion:	12.6.3
BuildVersion:	21G419
```

`python310` is installed from a third-party binary cache.

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-darwin"`
 - host os: `Darwin 21.6.0, macOS 10.16`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.13.3`
 - channels(maksbotan): `"nixpkgs"`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/Users/maksbotan/.nix-defexpr/channels/nixpkgs`
```

```console
$ nix-instantiate --eval -E '(import <nixpkgs> {}).lib.version'
"23.05pre480269.0d373d5af96"
```

**Comment by aequitas** [comment]
@maksbotan have you found a solution/workaround for this. I'm running into the same issue.

**Comment by maksbotan** [comment]
No, I installed python from python.org for those cases (fortunately they are rare for me).

**Comment by aequitas** [comment]
Thanks for the quick reply. My use case is for development environments managed by Devenv, so I want to avoid any external steps if can. I'll keep looking for a solution then.

**Comment by maksbotan** [comment]
I tried reading nixpkgs source code, AFAIU they need to update minimum apple sdk for whole nixpkgs, I'm not sure whether developers plan this in near future.

**Comment by aequitas** [comment]
@maksbotan I recently got reminded about this issue and I forgot the steps I needed to do to reproduce this issue for me. But it seems I'm not running into this issue anymore with our codebase. So maybe if that's the case for you, this issue can be closed. It might also be my codebase changed in some other way where this issue is no longer present for me.
